src/util/colorMap.ts

```diff
@@ -0,0 +1,58 @@
+import * as chroma from 'chroma-js';
+import * as vscode from 'vscode';
+
+export enum ColorType {
+	new = 'new',
+	fromRGB = 'fromRGB',
+	fromHSV = 'fromHSV'
+}
+
+type ExcludeSelfDoubleRecord<K extends string, V> = { [I in K]: Record<Exclude<K, I>, V> };
+export type ColorArray = [number, number, number];
+
+function clearNaN(value: Readonly<ColorArray>) {
+	return value.map(value => value !== value ? 0 : value) as ColorArray;
```

Comment:
Add a return type for the function `: ColorArray`, safer than doing a type assertion. (Also the next two functions as well)